Solen Ant Hole Add-On

DragonSlayer62 opened the pull request:
- [ADD] Added Solen Ants black and red
- [ADD] Added Solen Spawns for Felucca and Trammel
- [ADD] Added Solen hive teleporters
- [ADD] Added Solen Hive Dungeon Trammel Region
- [FIX] Fixed an issue with admin welcome gump where checkbox ID for decoration addons in Trammel were off by 1, resulting in wrong decorations being loaded
- [UPD] Resized admin welcome gump to accommodate additional decoration addons in Trammel

DragonSlayer62 committed:
Solen Ant Hole Add-On

This was introduced in LBR.
The script will allow the user to double click the middle black hole and teleport to the solen hive.

This will also spawn 2 to 5 ants when the hole is walked on.

Keeps track of the deaths of each ant so it cant over spawn the area.

DragonSlayer62 committed:
Solen Hive Main Entrance holes

These Holes do not move around or decay unlike the other hole that is meant to move around and decay and only take you into one spot.

This is only the Add-on for decoration no js script attached.

DragonSlayer62 committed:
Update

I added a 50% chance the hole doesn't do anything. according to all the information I found.

DragonSlayer62 committed:
Updated solen hives

Now we have the world decorations that will  decorate the main entrances around the map and the js script will work with these entrances and exits.

DragonSlayer62 committed:
small update to the teleporter

Added a timer to prevent spamming of npc ants.

Added another check make sure its not exit device so doesn't spawn ants.

DragonSlayer62 committed:
Update house.dfn

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
Update house.dfn

DragonSlayer62 committed:
Update to Teleport Script

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
Added new ant Npcs

DragonSlayer62 committed:
Update teleporter_solen_hive.js

DragonSlayer62 committed:
updates

DragonSlayer62 committed:
updates

DragonSlayer62 committed:
Update teleporter_solen_hive.js

DragonSlayer62 committed:
combined world decor

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
aos addition

DragonSlayer62 committed:
Merge branch 'solen-ant-hole' of https://github.com/DragonSlayer62/UOX3 into solen-ant-hole

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

strattond commented on data/dfndata/house/house.dfn:
```diff
@@ -3581,6 +3638,260 @@ Y=0
 }
 
+// dirt 1
+[HOUSE ITEM 600]
```
Is there a particular reason why the dirt ones are ever so slightly different in position?

DragonSlayer62 replied:
Yes they make a tunnel

<img width="327" height="290" alt="image" src="https://github.com/user-attachments/assets/d2061c31-047c-47e5-962a-31fb5700cbbd" />

<img width="235" height="193" alt="image" src="https://github.com/user-attachments/assets/3091fb98-63cf-4a11-aca7-31c5bc0f1fe7" />

strattond commented on data/dfndata/items/houseaddons/solenanthole.dfn:
```diff
@@ -0,0 +1,233 @@
+// solen ant hole bits and pieces
+
+[dirt1]
```
Worth a [basedirt] that has a name, weigh, movable and decay commonality?

DragonSlayer62 replied:
<img width="235" height="193" alt="image" src="https://github.com/user-attachments/assets/2ef2a8cd-2863-4d1b-a4fc-79849ef8a33d" />
<img width="210" height="219" alt="image" src="https://github.com/user-attachments/assets/2a8e2651-0ef0-4b7e-87a7-61e3b5e1d34a" />


this creates the tunnels two differant versions

strattond replied:
Yeah, my point was more that for dirt1 -> dirt9, the NAME, weight, movable and decay values are all the same.  It's only ID and possibly COLOUR that changes.

You could move the NAME, weight, movable and decay into a common basedirt and call it a day?

In fact, looking at solenentrancehole and it's variants, the weight, movable and decay values match there too.

So you could have a "base_immovable" that is the weight, movable and decay and GET those where appropriate.  For example:

```
[base_immovable]
{
weight=100
movable=2
decay=0
}

[ahole]
{
NAME=a hole
ID=0x122A
COLOUR=0x1
GET=base_immovable
SCRIPT=2752
]
```

DragonSlayer62 replied:
I see what you mean I fixed it.

strattond commented on data/dfndata/items/houseaddons/solenanthole.dfn:
```diff
@@ -0,0 +1,233 @@
+SCRIPT=2752
+}
+
+[shentrancepeice1]
```
Ditto for a [base_shentrancepiece] - also, you've spelt piece wrong :)

DragonSlayer62 replied:
will fix thanks

strattond commented on data/dfndata/npc/hymenoptera_base.dfn:
```diff
@@ -0,0 +1,373 @@
+
+[ambitious_black_solen_queen]
+{
+GETUO=ambitious_black_solen_queen_lbr
```
So the implication is that this LBR+ based on the name.

Should there be a GETUO, GETT2A, GETUOR or GETTD if they're prior?  What happens if they're missing?

Xoduz replied:
Originally for LBR+, but no reason why people shouldn't be able to enable it for earlier eras if desired. But the way the system is currently setup, it does require that tag for each era, or a broken NPC will spawn instead.

strattond commented on data/dfndata/npc/npc_aos/hymenoptera_aos.dfn:
```diff
@@ -0,0 +1,291 @@
+{
+GET=base_black_solen_warrior
+GOLD=250 350
+//Bracelet of Binding needs created
```
So ... where is it?  Either of these PACKITEMs don't seem a match?

DragonSlayer62 replied:
this not scripted yet comment for when it does get added

Xoduz replied:
I think you can comment out loot like this until the item functionality as been scripted. Don't add "junk items" that do nothing, can't be sold for money, just take up space, etc.

strattond commented on data/js/server/teleporter/teleporter_solen_hive.js:
```diff
@@ -0,0 +1,332 @@
+	"5728,1894": [5757, 1798],
+
+	//C leads to center area
+	"5657,1898": [5750, 1894],
```
So this is "C leads to center area", but 2 above with the same target coordinates is "B leads to level 2"

Are "center area" and "level 2" the same?

DragonSlayer62 replied:
https://[uo.stratics.com/content/atlas/solenhive.php](https://uo.stratics.com/content/atlas/solenhive.php) so the dungeon is just really one level but broken into 4 sections that have different teleport points.

strattond commented on data/js/server/teleporter/teleporter_solen_hive.js:
```diff
@@ -0,0 +1,332 @@
+*/
+
+// Entrance coordinates
+var entranceMap = {
```
Wondering if it's worth pulling targets out as meaningfully named constants and using those constants in the key value?
e.g. with the Solen Hive teleporters, there's lots that go to the same spot.

```
var SOLEN_LEVEL_2 = [5725, 1894];
...
"5757,1794": SOLEN_LEVEL_2,
```

DragonSlayer62 replied:
well, they go to the same area just different parts of that area

strattond reviewed (changes requested):
More questions than critiques

Xoduz commented on data/dfndata/spawn/trammel/spawn_trammel_dungeon_solen_hives.dfn:
```diff
@@ -0,0 +1,62 @@
+Y1=1781
+X2=5929
+Y2=2034
+WORLD=0
```
World number is incorrect on these spawn regions. By the name of the file they look intended for trammel?

DragonSlayer62 replied:
Fixed

DragonSlayer62 committed:
fixes requested

DragonSlayer62 committed:
changes made

changes made by request

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

Xoduz commented on data/dfndata/items/houseaddons/solenanthole.dfn:
If these items are intended specifically for the solen ant holes, rename them. Either add the dirt items as generic items in a more generic file, or rename them to be specifically for the solen ant holes. Example: [base_anthole_dirt], [anthole_dirt1], [anthole_dirt2], {anthole], etc.

DragonSlayer62 replied:
Ill rename them because there tied to a script

Xoduz commented on data/dfndata/spawn/felucca/spawn_felucca_dungeon_solen_hives.dfn:
```diff
@@ -0,0 +1,62 @@
+// spawn_felucca_dungeon_solen_hives.dfn
+// [4300-4349]
+
+[REGIONSPAWN 4300]
```
No. This is lazy design, and risks setting up situations where too many NPCs spawn in a single place, since it's all random across the entire dungeon. Split this up into additional spawn regions, 6 of them, each with MAXNPC=16 or some such. Same for Trammel spawns. Something like this:

<img width="807" height="610" alt="image" src="https://github.com/user-attachments/assets/14ff44fe-63c2-4ee6-b37d-9a2a2bc3d58a" />

Xoduz commented on data/js/jsdata/worldtemplates/felucca_solenhive.jsdata:
This is good stuff, but... it's not hooked up to anything, so these items will never spawn unless someone does 'decorate load felucca_solenhive

Ideally, this would be part of the admin welcome gump, as an option for additional decorations for Felucca/Trammel:

<img width="781" height="432" alt="image" src="https://github.com/user-attachments/assets/77698475-3ea6-4a97-97f4-3ce015e0229a" />


Also, the template file includes several "deed" items (id 5360), which spawn very visible in the world? I suggest removing these from the file.

Xoduz commented on data/js/server/teleporter/teleporter_solen_hive.js:
```diff
@@ -0,0 +1,332 @@
+
+		// Try entrance map first
+		if( teleportFromMap( pUser, key, entranceMap ))
+			return true;
```
"return true" here causes a system message that says "You can't think of a way to use that item." because it tries to pass the double-click event to code, and finds nothing actionable. Change to return false (as well as for the next one below.)

Xoduz commented on data/js/server/teleporter/teleporter_solen_hive.js:
```diff
@@ -0,0 +1,332 @@
+/*
+*	Solen Hives and Ants was introduced in LBR during a event called When Ants Attack.
+*
+*	We have the Randomly spawning black holes that can be put out with event tags for events.
```
Are these ant holes randomly spawned? Should there be spawn logic that makes this happen? Are they currently just fixed ant holes in every single possible location?

DragonSlayer62 replied:
the ones in decoration file are fixed the random ones is the ones a gm can use to put out for players to use and those will randomly drop you someone in the dungeon.

DragonSlayer62 committed:
Fixes Part 1

DragonSlayer62 committed:
changes requested

DragonSlayer62 committed:
Update spawn_trammel_dungeon_solen_hives.dfn

DragonSlayer62 committed:
update

DragonSlayer62 committed:
fixes

DragonSlayer62 committed:
Update teleporter_solen_hive.js

DragonSlayer62 committed:
Merge branch 'develop' into solen-ant-hole

DragonSlayer62 committed:
small fix

Xoduz committed:
Fix for admin welcome gump

- [FIX] Fixed an issue with admin welcome gump where checkbox ID for decoration addons in Trammel were off by 1, resulting in wrong decorations being loaded
- [UPD] Resized admin welcome gump to accommodate additional decoration addons in Trammel

Xoduz committed:
Merge branch 'develop' into solen-ant-hole
